eventbroadcaster/notifiersink.go

```diff
@@ -65,56 +65,63 @@ func (s *NotifierSink) Create(event *corev1.Event) (*corev1.Event, error) {
 }
 
 func (s *NotifierSink) getObject(event *corev1.Event) (client.Object, error) {
-	var obj client.Object
-
 	switch event.InvolvedObject.Kind {
 	case v1beta1.DisruptionKind:
-		disruption, err := s.getDisruption(event)
-		if err != nil {
-			s.logger.Warn(err)
-
-			// Unable to fetch the latest version of the disruption from the k8s api, we'll construct what we can from the event
-			disruption = v1beta1.Disruption{
-				TypeMeta: metav1.TypeMeta{
-					Kind:       event.InvolvedObject.Kind,
-					APIVersion: event.InvolvedObject.APIVersion,
-				},
-				ObjectMeta: metav1.ObjectMeta{
-					Name:            event.InvolvedObject.Name,
-					Namespace:       event.InvolvedObject.Namespace,
-					UID:             event.InvolvedObject.UID,
-					ResourceVersion: event.InvolvedObject.ResourceVersion,
-				},
+		var (
+			disruption v1beta1.Disruption
+			err        error
+		)
+
+		annotations := event.GetAnnotations()
+
+		// If the event has the annotation, use it to unmarshal the disruption
+		if disruptionString, ok := annotations[v1beta1.EventDisruptionAnnotation]; ok {
```

Comment:
We aren't setting this yet, could you implement this change in the disruption_webhook.go as well, please?

Reply:
I am now having a serious headache trying to remember, but in https://github.com/DataDog/chaos-controller/pull/892/files , I passed down the UserInfo in AnnotatedEvents, but I think I accidentally (or intentionally?) removed the commit that consumes that UserInfo annotation in the notifiersink.go.

So Disruptions still are missing UserInfo sometimes, even though I had a working patch, I think my rebases messed that up.

Thank you kindly, but could you please add to `disruption_webhook.go` the same changes you made to `disruption_cron_webhook.go`? 🙏